doc/index.md

```diff
@@ -125,6 +125,33 @@ vector with it:
     // vi == {1,2,2,3,3,3,4,4,4,4,5,5,5,5,5,...}
 ~~~~~~~
 
+### View constness
+
+View is a range, pair of iterators. In order to work, and work fast, many views need to cache some data.
+In order to keep iterators small sized - this cached data stored in the view itself, and iterators holds only pointers to their view.
+Because of cache, many views have non-const only `begin()`/`end()` (though they may return const-iterators).
+Const version of `begin()`/`end()` (where provided) - are truly const, hence - tread-safe.
+But since they can't cache - const versions usually slower then non-const.
```

Comment:
If `begin()` or `end()` needs to cache, they can't have a `const` overload because the `Range` concept imposes on them an amortized constant complexity requirement in http://wg21.link/range.range#3.2. So such slower versions are non-existent.

Reply:
@johelegp But it not says that const and non-const versions must have **exact** complexity and performance characteristics.

Imagine following.

`drop_exactly`. Currently, RandomAccess const and non-const versions are essentially the same. But if non-const version would cache (like !RandomAccess do now) - it would be faster for `std::deque`. 

```cpp
            // this is measurably faster for std::deque
            iterator_t<Rng> begin()
            {
                // copy-pasted from !RA version
                if(!begin_)
                    begin_ = next(ranges::begin(rng_), n_);
                return *begin_;
            }
            // fallback to slower version
            iterator_t<CRng> begin() const
            {
                return next(ranges::begin(rng_), n_);
            }
```

Both versions have amortized constant complexity.

Reply:
In the docs we have a lot to talk about, and the developer's time is precious. I'd like the docs to talk about complexity guarantees because they are important. I don't want the docs to lose that focus or make needless distinctions. What you say is not wrong, but I would leave it out in the interest of keeping the docs tightly focused.